primitives/src/witness.rs

```diff
@@ -185,37 +185,41 @@ impl Witness {
 
     /// Returns the last element in the witness, if any.
     #[inline]
-    pub fn last(&self) -> Option<&[u8]> {
-        if self.witness_elements == 0 {
-            None
-        } else {
-            self.nth(self.witness_elements - 1)
-        }
-    }
-
-    /// Returns the second-to-last element in the witness, if any.
-    #[inline]
-    pub fn second_to_last(&self) -> Option<&[u8]> {
-        if self.witness_elements <= 1 {
-            None
-        } else {
-            self.nth(self.witness_elements - 2)
-        }
-    }
+    pub fn last(&self) -> Option<&[u8]> { self.get_back(0) }
 
-    /// Returns the third-to-last element in the witness, if any.
-    #[inline]
-    pub fn third_to_last(&self) -> Option<&[u8]> {
-        if self.witness_elements <= 2 {
+    /// Retrieves an element from the end of the witness by its reverse index.
```

Comment:
Nice doc comment.  I wonder if this is the best place for the helper that needs to be exposed through the library as pub.

Reply:
This seems far more general than just applying to witness data.  I'm kinda surprised Rust doesn't have a way to do this without needing to build this method.

Reply:
Building it is not a big deal but I'm surprised slices don't have it already.